eng/Publishing.props

```diff
@@ -64,14 +66,20 @@
            the nupkgs pushed. Do not do this if building from source, since we want the source build intermediate package
            to be published. Use ArcadeBuildFromSource as DotNetBuildFromSource is only set in the internal source build,
            and Build.proj is invoked from the wrapper build. -->
-      <ItemsToPushToBlobFeed Remove="@(ItemsToPushToBlobFeed)" Condition="'$(OS)' != 'Windows_NT' and '$(PostBuildSign)' != 'true' and '$(ArcadeBuildFromSource)' != 'true'" />
+      <ItemsToPushToBlobFeed Remove="@(ItemsToPushToBlobFeed)" Condition="'$(OS)' != 'Windows_NT' and '$(PostBuildSign)' != 'true' and '$(DotNetBuildRepo)' != 'true'" />
 
       <ItemsToPushToBlobFeed Include="@(_ChecksumsToPublish)">
         <ManifestArtifactData>NonShipping=true</ManifestArtifactData>
         <PublishFlatContainer>true</PublishFlatContainer>
         <RelativeBlobPath>$(_UploadPathRoot)/Runtime/$(_PackageVersion)/%(Filename)%(Extension)</RelativeBlobPath>
       </ItemsToPushToBlobFeed>
 
+      <ItemsToPushToBlobFeed Include="@(_VersionFileToPublish)">
```

Comment:
Was this not getting published before?

Reply:
Not in VMR build, no.

Reply:
OK - I see why - this is conditioned on `PublishInstallerBaseVersion` - https://github.com/dotnet/aspnetcore/blob/d9df235c2e6d41b65d8e860c5e209d9df8cbcae1/eng/Publishing.props#L34-L35

The property is only set in CI: https://github.com/dotnet/aspnetcore/blob/d9df235c2e6d41b65d8e860c5e209d9df8cbcae1/.azure/pipelines/ci.yml#L350

I should probably update the existing condition to also publish in VMR (i.e. when '$(DotNetBuildRepo)' == 'true')?

Reply:
Yes. When you see things like this, e.g. `OS != Windows_NT` or some kind of random flag that says `SkipPublishingInstallers` or something along those lines, it's that the repo's official build was publishing duplicates of assets between two legs. So Windows and OSX both produced Foo.Bar. The rule in our official builds is that when the merged manifest is generated, you can't have any duplicates from the input manifests.

This is not the rule in the VMR and we don't want to do this here. The asset selection that will happen at the merge point will be responsible for deciding which of a set of duplicates wins, and we'll add metadata to repo builds as necessary to feed the algorithm. This is in contrast to the official repo builds, which just avoid publishing altogether for duplicate assets.

So, much of your publishing.props changes in the repos can get a lot simpler (maybe go away altogether)

Reply:
Fixed with https://github.com/dotnet/aspnetcore/pull/54422/commits/9dc87deac025a7ca40b4fa21c542afc497e11cc8